.github/templates/main.yml

```diff
@@ -66,11 +66,12 @@ steps:
       retention-days: 1
 #@ end
 
-#@ def checkoutCode():
+#@ def checkoutCode(submodules=False):
   - name: Checkout code
     uses: actions/checkout@v2
     with:
-      submodules: recursive
+      submodules: #@ submodules
+      ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
I want to double check that I got this right:
Our main workflow is triggered with push on master/main and on pull_request (with events _opened, synchronize, or reopened_). Since `ref` defaults to the reference or SHA of the event that triggered the workflow, you're saying that we need to avoid to fetch a push to master, which generally happens to be a merge of a pull request, but we always need to carefully fetch the head of the current PR which could be lacking the latest merged PRs that master has.
Did I get this right?

Reply:
The default behavior for checkout on a PR trigger is to checkout the merge commit (i.e. have the PR branch merged into the base branch and use that). This is why when there are merge conflicts, our GHA workflows don't run. This is the change suggested in the [checkout action docs](https://github.com/actions/checkout#Checkout-pull-request-HEAD-commit-instead-of-merge-commit) to make sure that PR builds actually build and test the code that is in the PR itself.

While for PR builds against master, checking out the merge commit is not a big deal - and in some cases may even be useful - for release PRs it's plain wrong and may result in very surprising behavior where we actually push a nuget package containing code that was not present in the release branch.

Reply:
Now it all makes sense. I wasn't fully aware of the default behaviour for the checkout on PRs, that's why I got confused. Thank you for the explanation. On top of this, the change of line 74 will stop "there are merge conflicts with master" that prevent the workflow to run.